app/models/ott/CategorisationInfo.scala

```diff
@@ -102,23 +105,32 @@ object CategorisationInfo {
         val category2Assessments = assessmentsSorted.filter(ass => ass.isCategory2)
 
         val category1ToAnswer = category1Assessments.filter(ass => !ass.hasNoAnswers).filter(ass => !ass.isNiphlsAnswer)
-        val category2ToAnswer = category2Assessments.filter(ass => !ass.hasNoAnswers)
+        val category2ToAnswer = category2Assessments.filter(ass => !ass.hasNoAnswers).filter(ass => !ass.isNirmsAnswer)
 
         val areAllCategory1Answerable = category1ToAnswer.size == category1Assessments.size
         val areAllCategory2Answerable = category2ToAnswer.size == category2Assessments.size
 
         val isNiphlsAssessment =
           category1Assessments.exists(ass => ass.isNiphlsAnswer) && category2Assessments.exists(ass => ass.hasNoAnswers)
 
+        val isNirmsAssessment = category2Assessments.exists(ass => ass.isNirmsAnswer)
+
+        val isTraderNiphlAuthorised = traderProfile.niphlNumber.isDefined
+        val isTraderNirmsAuthorised = traderProfile.nirmsNumber.isDefined
+
         val questionsToAnswer = {
-          if (isNiphlsAssessment && traderProfile.niphlNumber.isDefined) {
+          if (isNiphlsAssessment && isTraderNiphlAuthorised) {
             category1ToAnswer
           } else if (isNiphlsAssessment) {
             Seq.empty
           } else if (!areAllCategory1Answerable) {
             Seq.empty
           } else if (!areAllCategory2Answerable) {
-            category1ToAnswer
+            if (isNirmsAssessment && isTraderNirmsAuthorised) {
```

Comment:
this could probably be rewritten without nesting

```         if (isNiphlsAssessment && isTraderNiphlAuthorised) {
            category1ToAnswer
          } else if (isNiphlsAssessment) {
            Seq.empty
          } else if (!areAllCategory1Answerable) {
            Seq.empty
          } else if (isNirmsAssessment && isTraderNirmsAuthorised) {
              category1ToAnswer ++ category2ToAnswer
         } else if (!areAllCategory2Answerable) {
             category1ToAnswer
          } else {
            category1ToAnswer ++ category2ToAnswer
          }
            ```

Reply:
Updated